crates/evm/src/evm/system_contracts/mod.rs

```diff
@@ -58,20 +72,22 @@ impl Bridge {
         address!("3100000000000000000000000000000000000002")
     }
 
-    pub(crate) fn initialize(data: Vec<u8>) -> Bytes {
-        let mut func_selector: Vec<u8> = vec![0x45, 0x07, 0xfc, 0x69]; // initialize(bytes,bytes,uint256,address) 4507fc69
-
-        func_selector.extend_from_slice(data.as_slice());
-
-        Bytes::from(func_selector)
+    pub(crate) fn initialize() -> Bytes {
+        // Hardcoded until better times.
+        BridgeContract::initializeCall {
+            _depositScript: hex!("d2205daf577048c5e5a9a75d0a924ed03e226c3304f4a2f01c65ca1dab73522e6b8bad206228eba653cf1819bcfc1bc858630e5ae373eec1a9924322a5fe8445c5e76027ad201521d65f64be3f71b71ca462220f13c77b251027f6ca443a483353a96fbce222ad200fabeed269694ee83d9b3343a571202e68af65d05feda61dbed0c4bdb256a6eaad2000326d6f721c03dc5f1d8817d8f8ee890a95a2eeda0d4d9a01b1cc9b7b1b724dac00630663697472656114").into(),
+            _scriptSuffix: hex!("0800000000000f424068").into(),
+            _requiredSigsCount: U256::from(5),
+            _owner: address!("f9725b63fe14efaf7cc705ba4e5c55a03d50e940"),
+        }
+        .abi_encode()
+        .into()
     }
 
     /// Return data to deposit
-    pub fn deposit(data: Vec<u8>) -> Bytes {
-        let mut func_selector: Vec<u8> = vec![0xdd, 0x95, 0xc7, 0xc6]; // deposit((bytes4,bytes2,bytes,bytes,bytes,bytes4,bytes,uint256,uint256)) dd95c7c6
-
-        func_selector.extend_from_slice(data.as_slice());
-
-        Bytes::from(func_selector)
+    pub fn deposit(params: BridgeContract::DepositParams) -> Bytes {
```

Comment:
all these changes on this file from a readability perspective is 1000x better.

however, how do you think the performance is effected?

don't forget this part runs in zk context.